lib/image.js

```diff
@@ -1,12 +1,84 @@
-"use strict";
-var Promise = require('bluebird');
-var fs = require('fs-extra');
+'use strict';
+var Cesium = require('cesium');
+var fsExtra = require('fs-extra');
 var path = require('path');
+var PNG = require('pngjs').PNG;
+var Promise = require('bluebird');
+
+var fsExtraReadFile = Promise.promisify(fsExtra.readFile);
 
-var fsReadFile = Promise.promisify(fs.readFile);
+var defaultValue = Cesium.defaultValue;
+var WebGLConstants = Cesium.WebGLConstants;
 
 module.exports = loadImage;
 
+/**
+ * Load an image file and get information about it.
+ *
+ * @param {String} imagePath Path to the image file.
+ * @param {Object} [options] An object with the following properties:
+ * @param {Boolean} [options.hasTransparency=false] Do a more exhaustive check for texture transparency by looking at the alpha channel of each pixel.
+ * @returns {Promise} A promise resolving to the image information, or undefined if the file doesn't exist.
+ *
+ * @private
+ */
+function loadImage(imagePath, options) {
+    options = defaultValue(options, defaultValue.EMPTY_OBJECT);
+    var hasTransparency = defaultValue(options.hasTransparency, false);
+
+    return fsExtraReadFile(imagePath)
+        .then(function(data) {
+            var extension = path.extname(imagePath);
+
+            var info = {
+                transparent : false,
+                format : getFormat(3),
+                source : data,
+                extension : extension
+            };
+
+            if (extension === '.png') {
```

Comment:
Never do string comparisons like this, what about `.PNG` or any other combination of case? Use a regular expression instead, in this case `/\.png/i.test(extension)` handles everything.